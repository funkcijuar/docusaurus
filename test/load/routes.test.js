import genRoutesConfig from '@lib/load/routes';
import loadDocs from '@lib/load/docs';
import loadPages from '@lib/load/pages';
import path from 'path';

describe('genRoutesConfig', () => {
  const pagesDir = path.join(__dirname, '__fixtures__', 'simple-pages');
  const docsDir = path.join(__dirname, '__fixtures__', 'simple-docs');

  test('website with pages and docs', async () => {
    const props = {
      docsData: await loadDocs(docsDir),
      pagesData: await loadPages(pagesDir)
    };
    const routes = await genRoutesConfig(props);
    expect(routes).toMatchSnapshot();
  });

  test('website with only pages', async () => {
    const props = {
      pagesData: await loadPages(pagesDir)
    };
    const routes = await genRoutesConfig(props);
    expect(routes).toMatchSnapshot();
  });

  test('website with only docs', async () => {
    const props = {
      docsData: await loadDocs(docsDir)
    };
    const routes = await genRoutesConfig(props);
    expect(routes).toMatchSnapshot();
  });

  test('website with no docs/pages', async () => {
    const routes = await genRoutesConfig({});
    expect(routes).toMatchSnapshot();
  });
});
