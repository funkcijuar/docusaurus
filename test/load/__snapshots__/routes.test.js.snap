// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genRoutesConfig website with no docs/pages 1`] = `
"import React from 'react';
import Loadable from 'react-loadable';
import Loading from '@theme/Loading';
import Docs from '@theme/Docs';
import NotFound from '@theme/NotFound';
const routes = [,,
  {
    path: '*',
    component: NotFound
  }
];
export default routes;
"
`;

exports[`genRoutesConfig website with only docs 1`] = `
"import React from 'react';
import Loadable from 'react-loadable';
import Loading from '@theme/Loading';
import Docs from '@theme/Docs';
import NotFound from '@theme/NotFound';
const routes = [
  {
    path: \\"/hello\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/hello.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },
  {
    path: \\"/foo/bar\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/foo/bar.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },
  {
    path: \\"/foo/baz\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/foo/baz.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },,
  {
    path: '*',
    component: NotFound
  }
];
export default routes;
"
`;

exports[`genRoutesConfig website with only pages 1`] = `
"import React from 'react';
import Loadable from 'react-loadable';
import Loading from '@theme/Loading';
import Docs from '@theme/Docs';
import NotFound from '@theme/NotFound';
const routes = [,
  {
    path: \\"/foo\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/foo.js'),
      loading: Loading
    })
  },
  {
    path: \\"/\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/index.js'),
      loading: Loading
    })
  },
  {
    path: \\"/bar/baz\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/bar/baz.js'),
      loading: Loading
    })
  },
  {
    path: \\"/foo/\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/foo/index.js'),
      loading: Loading
    })
  },
  {
    path: '*',
    component: NotFound
  }
];
export default routes;
"
`;

exports[`genRoutesConfig website with pages and docs 1`] = `
"import React from 'react';
import Loadable from 'react-loadable';
import Loading from '@theme/Loading';
import Docs from '@theme/Docs';
import NotFound from '@theme/NotFound';
const routes = [
  {
    path: \\"/hello\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/hello.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },
  {
    path: \\"/foo/bar\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/foo/bar.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },
  {
    path: \\"/foo/baz\\",
    exact: true,
    component: Loadable({
      loader: () => import('@docs/foo/baz.md'),
      loading: Loading,
      render(loaded, props) {
        let Content = loaded.default;
        return <Docs {...props}><Content /></Docs>;
      }
    })
  },
  {
    path: \\"/foo\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/foo.js'),
      loading: Loading
    })
  },
  {
    path: \\"/\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/index.js'),
      loading: Loading
    })
  },
  {
    path: \\"/bar/baz\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/bar/baz.js'),
      loading: Loading
    })
  },
  {
    path: \\"/foo/\\",
    exact: true,
    component: Loadable({
      loader: () => import('@pages/foo/index.js'),
      loading: Loading
    })
  },
  {
    path: '*',
    component: NotFound
  }
];
export default routes;
"
`;
